compendium/DeclarativeServices/src/manager/states/CCActiveState.cpp

```diff
@@ -102,64 +117,84 @@ void CCActiveState::DoDeactivateWork(ComponentConfigurationImpl& mgr)
 
 bool CCActiveState::Modified(ComponentConfigurationImpl& mgr)
 {
-  std::lock_guard<std::mutex> lock(oneAtATimeMutex);
-  // Make sure the state didn't change while we were waiting
-  if (mgr.GetState()->GetValue() !=
-      service::component::runtime::dto::ComponentState::ACTIVE) {
-    auto logger = mgr.GetLogger();
-    logger->Log(cppmicroservices::logservice::SeverityLevel::LOG_WARNING,
-                "Modified failed. Component no longer in Active State.");
-    return false;
-  }
-  if (!mgr.ModifyComponentInstanceProperties()) {
-    // Component instance does not have a Modified method. Deactivate
-    // and reactivate
-    DoDeactivateWork(mgr);
-    // Service registration properties will be updated when the service is
-    // registered. Don't need to do it here.
-    return false;
+  auto logger = mgr.GetLogger();
+  if (latch.CountUp()) {
+    detail::ScopeGuard sg([this, logger]() {
+      // By using try/catch here, we ensure that this lambda function doesn't
+      // throw inside LatchScopeGuard's dtor.
+      try {
+        latch.CountDown();
+      } catch (...) {
+        logger->Log(cppmicroservices::logservice::SeverityLevel::LOG_ERROR,
+                    "latch.CountDown() threw an exception during "
+                    "LatchScopeGuard cleanup in CCActiveState::Modified.",
+                    std::current_exception());
+      }
+    });
+    std::lock_guard<std::mutex> lock(oneAtATimeMutex);
+    // Make sure the state didn't change while we were waiting
+    auto currentState = mgr.GetState();
+    if (currentState->GetValue() !=
+        service::component::runtime::dto::ComponentState::ACTIVE) {
+      auto logger = mgr.GetLogger();
+      logger->Log(cppmicroservices::logservice::SeverityLevel::LOG_WARNING,
+                  "Modified failed. Component no longer in Active State.");
+      return false;
+    }
+    if (!mgr.ModifyComponentInstanceProperties()) {
+      // Component instance does not have a Modified method. Deactivate
+      // and reactivate
+      latch.CountDown();
```

Comment:
this should be wrapped in a `try/catch` to guarantee that deactivation happens even if the latch throws.

Reply:
Yes. Fixed.

Reply:
this still needs to be wrapped in a `try/catch`.